### PR TITLE
Add loading indicator text when loading large code files

### LINKED ISSUE #44666: FE scale audit: loading big files in search results list bogs down the browser - strategic readiness
### Description
According to feedback from one of our [customers](https://github.com/sourcegraph/accounts/issues/6716) ([slack thread](https://sourcegraph.slack.com/archives/C01P4R0RZH9/p1668701456260109) for more context) the large files bog down the browser while doing syntax highlighting and rendering.

### Plan

- [ ] Research the described behavior (so far @philipp-spiess hasn't tested with files that big for the FE scale audit yet)
- [ ] Understand if using codemirror will solve the problem
- [ ] If not, explore potential UI solutions (ex. the download button suggested by the customer) - potential design input needed

### PR DESCRIPTION
**Pull request by philipp-spiess**
Fixes #44666

When opening large files, the file download can become quite slow and since we don't use any streaming APIs, [the bottleneck is often the time to first byte](https://github.com/sourcegraph/sourcegraph/issues/44666#issuecomment-1330789819). To improve the UX in the short term before we can [add larger scale fixes](https://github.com/sourcegraph/sourcegraph/issues/44666#issuecomment-1330796505) to this problem, this PR tweaks the loading experience and adds different subtitles when loading exceeds specific thresholds (1sec, 3sec, 6sec, and 9sec). https://www.figma.com/file/5CgqjaXqrRhquTG6GB3YGd/FE-scale-audit%3A-loading-big-files-in-search-results-list-bogs-down-the-browser-%2344666-WIP?node-id=275%3A1628&t=TaPSg9bCLwiM1A1N-4

Figma: https://www.figma.com/file/5CgqjaXqrRhquTG6GB3YGd/FE-scale-audit%3A-loading-big-files-in-search-results-list-bogs-down-the-browser-%2344666-WIP?node-id=275%3A1628&t=TaPSg9bCLwiM1A1N-4

## Test plan

You have to enable heavy network throttling to repro the 9 sec case (100kbps in my case):

https://user-images.githubusercontent.com/458591/205664147-c5ac41a3-d561-4e0b-96d7-70d3304a6f4b.mov

## App preview:

- [Web](https://sg-web-ps-large-file-loading.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+0.97 kb) | 0.01% (+0.97 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/large-file-loading/compare-report.html;tab=live_object) for a full comparison between the commits 522decc and e70be36 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by AlicjaSuska — APPROVED** [review]
Thank you for working on this @philipp-spiess Looks awesome!

File: client/web/src/repo/blob/BlobLoadingSpinner.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            {afterOneSec ? (
+                <div className="d-flex flex-column align-items-center">
+                    <LoadingSpinner />
+                    <div className="text-muted mt-2">
+                        {afterNineSec ? (
+                            <>
+                                It’s taking much longer than expected to load this file. Try{' '}
+                                <Link to={location.pathname} onClick={reload}>
+                                    reloading the page
+                                </Link>
```
Comment (lrhacker) [inline client/web/src/repo/blob/BlobLoadingSpinner.tsx:43]:
```suggestion
                                <Link to={location.pathname} onClick={reload}>
                                    reloading the page
                                </Link>.
```

Adding a period here so both statements are consistent looks more natural to me.

**Review by lrhacker — APPROVED** [review]
Nice and simple improvement until bigger changes can be made. 👍

**Pull request merged**